src/quick-lint-js/fe/parse.cpp

```diff
@@ -734,6 +734,55 @@ void parser::consume_semicolon() {
   }
 }
 
+void parser::error_on_pointless_nullish_coalescing_operator(
+    expression::binary_operator* ast) {
+  
+  auto is_nullish_operator = [](string8_view s) -> bool {
+    return s == u8"??"_sv;
+  };
+
+  auto no_left_operand = [](string8_view s) -> bool {
+    return s == u8"??"_sv;
+  };
+
+  if (no_left_operand(ast->span().string_view())){
+    return;
+  }
```

Comment:
test_parse_expression_statement.statement_starting_with_binary_only_operator failed if I don't include this check. This makes sure we don't report the nullish diagnostic when the expression is missing a left operand since its more relevant to report that the lhs is missing rather than it is always null. However, changing from blacklist to whitelist should make this check unnecessary.